src/Entity/OgMembership.php

```diff
@@ -203,7 +203,7 @@ public function getRoles() {
   public function setRoles(array $roles = []) {
     $role_ids = array_map(function (OgRole $role) {
       return $role->id();
-    }, $roles);
+    }, array_values($roles));
```

Comment:
as discussed, please check why `array_values()` is needed here, because it looks a bit strange.